impeller/renderer/backend/vulkan/command_encoder_vk_unittests.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <thread>
+
+#include "flutter/fml/synchronization/count_down_latch.h"
+#include "flutter/testing/testing.h"
+#include "impeller/renderer/backend/vulkan/command_encoder_vk.h"
+#include "impeller/renderer/backend/vulkan/fence_waiter_vk.h"
+#include "impeller/renderer/backend/vulkan/test/mock_vulkan.h"
+
+namespace impeller {
+namespace testing {
+
+TEST(CommandEncoderVKTest, DeleteEncoderAfterThreadDies) {
```

Comment:
I have sort of mixed feelings about these tests, because its sort disjointed from how these classes are used. I would feel better if there was substantially more commentary on specifically what these tests are covering.

Reply:
Or at least for one of them, and then subsequent tests could have less commentary.

Reply:
Unit tests are supposed to be testing how a class can be used, not how they are supposed to be used.  Maybe it will help to understand why I implemented this fix.  The pool change is a fix for a real bug that I was running into when using the playgrounds test harness which Chinmay was trying to get working.  So, it may be true that this bug shouldn't present itself in production, it has demonstrably happened in our test harness and may present itself accidentally in production.

The second fix with the resourcemanager is definitely something that can manifest in production code, but was showing up regularly in these tests.

We typically don't annotate unit tests that are using the public api of a method.  I'm happy to add any comments you think will help.

Reply:
I understand that, what I'm saying is that I'm having a hard time following the test and so I'd appreciate more commentary on how/why the particular test successfully exercises the assertions.

Reply:
I added a comment to the first test, expanded the second comment and removed one piece of vestigial code.